pkg/controllers/state/cluster.go

```diff
@@ -51,11 +51,12 @@ type Cluster struct {
 	cloudProvider cloudprovider.CloudProvider
 	clock         clock.Clock
 
-	mu               sync.RWMutex
-	nodes            map[string]*StateNode           // provider id -> cached node
-	bindings         map[types.NamespacedName]string // pod namespaced named -> node name
-	nameToProviderID map[string]string               // node name -> provider id
-	daemonSetPods    sync.Map                        // daemonSet -> existing pod
+	mu                       sync.RWMutex
+	nodes                    map[string]*StateNode           // provider id -> cached node
+	bindings                 map[types.NamespacedName]string // pod namespaced named -> node name
+	nodeNameToProviderID     map[string]string               // node name -> provider id
+	nodeClaimKeyToProviderID map[nodeclaimutil.Key]string    // node claim key -> provider id
```

Comment:
IIUC the NodeClaim key is just a transitionary representation when we're moving from Machine to NodeClaims. After the transition, NodeClaims should just be a unique string name?